src/main/java/org/folio/circulation/domain/RequestRepository.java

```diff
@@ -82,6 +82,17 @@ public static RequestRepository using(Clients clients, boolean fetchMaterialType
       .thenComposeAsync(result -> result.after(patronGroupRepository::findPatronGroupsForRequestsUsers));
   }
 
+  public CompletableFuture<Result<MultipleRecords<Request>>> findByWithRelatedRecords(
```

Comment:
The existing `findBy` method already fetches the related records, I think it makes sense to make this an overload.

Also, please consolidate the implementation if possible.

```suggestion
  public CompletableFuture<Result<MultipleRecords<Request>>> findBy(
```

Reply:
Done